packages/core/src/components/tag-input/tagInput.tsx

```diff
@@ -414,4 +415,15 @@ export class TagInput extends AbstractPureComponent<ITagInputProps, ITagInputSta
     private isValidIndex(index: number) {
         return index !== NONE && index < this.props.values.length;
     }
+
+    private handlePaste = (event: React.ClipboardEvent<HTMLInputElement>) => {
+        // we don't want it to show up in the input since we're directly adding the terms
+        event.preventDefault();
+        const { onChange, values } = this.props;
+        const clipboardText = event.clipboardData.getData("text");
+        const newValues = [].concat(...clipboardText.split(/\n|\r/).map(s => this.getValues(s)));
```

Comment:
ok hold up here's how this PR should look: `onPaste` should trigger the same code path as pressing "enter" in the input. we should change the separator default to include newlines.